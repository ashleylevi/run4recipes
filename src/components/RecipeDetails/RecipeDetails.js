import React from 'react';
import { Link, withRouter, Redirect } from 'react-router-dom';
import { connect } from 'react-redux'

export const RecipeDetails = (props) => {
  let category;
  if(props.pastaRecipes.length) {
    if (props.match.path.includes('pasta')) {
      category = props.pastaRecipes
    } else if (props.match.path.includes('potato')) {
      category = props.potatoRecipes
    } else if (props.match.path.includes('bread')) {
      category = props.breadRecipes
    } else {
      return 
    }
  } else {
    return (<Redirect to='/'/>)
  }
  const recipe = category.find((pasta) => {
    return pasta.name === props.match.params.name
  })
    
  const ingredients = recipe.ingredients.map((ingredient) => {
    return ingredient
  })

  return (
    <div>
      <h1>{recipe.name}</h1>
      <img src={recipe.image}/>
      <div>{ingredients}</div>

      {/* <Link to={`/${name}`} className='back-btn'>â—€ back</Link> */}
      {/* <img src={image} className='app-img-no-hover'/>
      <p className='creature-bio'>{bio}</p> */}
    </div>
  )
}

const mapStateToProps = state => ({
  pastaRecipes: state.pastaRecipes,
  potatoRecipes: state.potatoRecipes,
  breadRecipes: state.breadRecipes
})

export default withRouter(connect(mapStateToProps)(RecipeDetails))