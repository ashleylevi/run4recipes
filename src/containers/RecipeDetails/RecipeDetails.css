.recipe-title {
  text-transform: uppercase;
  font-size: 20px;
  text-align: center;
  color: #32636E;
  margin-top: 40px;
  font-weight: 700;
}

.recipe-div {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: 40px 0 40px 0;
}

.img {
  border: 1px solid black;
  height: 400px;
  width: 300px;
}

li {
  font-family: Raleway;
  font-weight: bold;
  font-size: 19px;
  color: #3D323E;
}

ul {
  margin-top: 0;
}
