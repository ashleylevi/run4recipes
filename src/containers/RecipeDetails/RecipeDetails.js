import React from 'react';
import { withRouter, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './RecipeDetails.css';

export const RecipeDetails = (props) => {
  let category;
  if(props.pastaRecipes.length) {
    if (props.match.path.includes('pasta')) {
      category = props.pastaRecipes
    } else if (props.match.path.includes('potato')) {
      category = props.potatoRecipes
    } else if (props.match.path.includes('bread')) {
      category = props.breadRecipes
    } else {
      return 
    }
  } else {
    return (<Redirect to='/'/>)
  }
  const recipe = category.find((obj) => {
    return obj.name === props.match.params.name
  })
    
  const ingredients = recipe.ingredients.map((ingredient) => {
    return <li><p className="ingredient">{ingredient}</p></li>
  })

  return (
    <div className="ingredients-div">
      <h1 className="recipe-title">{recipe.name}</h1>
      <div className="recipe-div">
        <img src={recipe.image} className="img"/>
        <ul className="recipe-ingredients">{ingredients}</ul>
      </div>
    </div>
  )
}

export const mapStateToProps = state => ({
  pastaRecipes: state.pastaRecipes,
  potatoRecipes: state.potatoRecipes,
  breadRecipes: state.breadRecipes
})

export default withRouter(connect(mapStateToProps)(RecipeDetails))

RecipeDetails.propTypes = {
  pastaRecipes: PropTypes.array,
  potatoRecipes: PropTypes.array,
  breadRecipes: PropTypes.array
}