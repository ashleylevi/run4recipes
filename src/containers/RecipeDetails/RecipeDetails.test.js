import { mapStateToProps } from './RecipeDetails';
import { shallow } from 'enzyme';
import { RecipeDetails } from './RecipeDetails';
import React from 'react';

describe('RecipeDetails', () => {
  describe('RecipeDetails component', () => {
    const mockRecipes  = [{
          label: "Spaghetti and meatballs",
          image: "https://image.png",
          ingredientLines: [],
          category: 'bread'
    }]
    const mockPath = {
      path: ''
    }

    it('Should match the snapshot', () => {
      const wrapper = shallow(<RecipeDetails pastaRecipes={mockRecipes} potatoRecipes={mockRecipes} breadRecipes={mockRecipes} match={mockPath} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
  
  describe('mapStateToProps', () => {
    it('should return an array with recipe objects', () => {
      const mockRecipes  = [{
        label: "Spaghetti and meatballs",
        image: "https://image.png",
        ingredientLines: [],
        category: 'bread'
      }]
    
      const mockState = {
        pastaRecipes: mockRecipes,
        potatoRecipes: mockRecipes,
        breadRecipes: mockRecipes
      }

      const expected = {
        pastaRecipes: mockRecipes,
        potatoRecipes: mockRecipes,
        breadRecipes: mockRecipes
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})