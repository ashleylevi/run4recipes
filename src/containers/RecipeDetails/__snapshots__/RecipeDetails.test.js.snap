// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetails RecipeDetails component Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeDetails
    breadRecipes={
      Array [
        Object {
          "category": "bread",
          "image": "https://image.png",
          "ingredientLines": Array [],
          "label": "Spaghetti and meatballs",
        },
      ]
    }
    match={
      Object {
        "path": "",
      }
    }
    pastaRecipes={
      Array [
        Object {
          "category": "bread",
          "image": "https://image.png",
          "ingredientLines": Array [],
          "label": "Spaghetti and meatballs",
        },
      ]
    }
    potatoRecipes={
      Array [
        Object {
          "category": "bread",
          "image": "https://image.png",
          "ingredientLines": Array [],
          "label": "Spaghetti and meatballs",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
